src/lib/router.ts

```diff
@@ -21,10 +21,56 @@ router.post('/matches', async (req, res, next) => {
 
     const inserted = await getMatchesCollection().insertOne(match);
     if (!inserted.acknowledged) {
-      res.status(500).send('Error inserting match');
+      response.status(500).send('Error inserting match');
       return;
     }
-    res.status(200).json(match);
+    response.status(200).json(match);
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.post('/matches/:gameId/highlights', async (request, response, next) => {
```

Comment:
You mistake the `gameId` for the `matchId`. The `gameId` is not unique and is for e.g. League of Legends.
Thus, the ID here should target the specific match. I would use the MongoDB `_id` to have a call like:
`/matches/60f8196b0cc5b81d77c551d3/highlights` where `60f8196b0cc5b81d77c551d3` is the String of `_id`.